freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/PasswordPolicyService.java

```
@@ -0,0 +1,77 @@
+package com.sequenceiq.freeipa.service.freeipa.flow;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.PasswordPolicy;
+
+@Service
+class PasswordPolicyService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PasswordPolicyService.class);
+
+    private static final String MIN_PASSWORD_LENGTH = "krbpwdminlength";
+
+    private static final Integer MIN_PASSWORD_LENGTH_VALUE = 8;
+
+    private static final String MIN_CHARACTER_CLASSES = "krbpwdmindiffchars";
+
+    private static final Integer MIN_CHARACTER_CLASSES_VALUE = 3;
+
+    private static final String MAX_PASSWORD_LIFETIME_DAYS = "krbmaxpwdlife";
+
+    private static final Integer MAX_PASSWORD_LIFETIME_DAYS_VALUE = 90;
+
+    private static final String MAX_FAILURES_BEFORE_LOCK = "krbpwdmaxfailure";
+
+    private static final Integer MAX_FAILURES_BEFORE_LOCK_VALUE = 10;
+
+    private static final String FAILURE_RESET_INTERVALL_SECONDS = "krbpwdfailurecountinterval";
+
+    private static final Integer FAILURE_RESET_INTERVALL_SECONDS_VALUE = 3;
+
+    private static final String LOCKOUT_DURATION_SECONDS = "krbpwdlockoutduration";
+
+    private static final Integer LOCKOUT_DURATION_SECONDS_VALUE = 10;
+
+    void updatePasswordPolicy(FreeIpaClient freeIpaClient) {
+        try {
+            PasswordPolicy config = freeIpaClient.getPasswordPolicy();
+            Map<String, Object> params = getParameters(config);
+            if (!params.isEmpty()) {
+                freeIpaClient.updatePasswordPolicy(params);
+            }
+        } catch (FreeIpaClientException e) {
+            LOGGER.error("Failed to create policy.", e);
```

we should throw the error here to make freeipa provision fail